e2e/src/common/utils.ts

```diff
@@ -170,34 +171,63 @@ export async function sendXTransactions(
   }
 }
 
-export async function sendTransactionsToGenerateTrafficWithInterval(
-  signer: Wallet,
-  pollingInterval: number = 1000,
-) {
-  const [maxPriorityFeePerGas, maxFeePerGas] = getAndIncreaseFeeData(await signer.provider.getFeeData());
+export async function sendTransactionsToGenerateTrafficWithInterval(signer: Wallet, pollingInterval: number = 1000) {
+  const [maxPriorityFeePerGas, maxFeePerGas] = getAndIncreaseFeeData(await signer.provider!.getFeeData());
   const transactionRequest = {
     to: signer.address,
-    value: ethers.utils.parseEther("0.000001"),
+    value: etherToWei("0.000001"),
     maxPriorityFeePerGas: maxPriorityFeePerGas,
     maxFeePerGas: maxFeePerGas,
-  }
+  };
 
-  return setInterval(async function () {
-    const tx = await signer.sendTransaction(transactionRequest);
-    await tx.wait();
-  }, pollingInterval);
+  let timeoutId: NodeJS.Timeout | null = null;
+  let isRunning = true;
+
+  const sendTransaction = async () => {
+    if (!isRunning) return;
+
+    try {
+      const tx = await signer.sendTransaction(transactionRequest);
+      await tx.wait();
+
+      if (isRunning) {
+        timeoutId = setTimeout(sendTransaction, pollingInterval);
+      }
+    } catch (error) {
+      console.error("Error sending transaction:", error);
+      if (isRunning) {
+        timeoutId = setTimeout(sendTransaction, pollingInterval);
+      }
+    }
```

Comment:
Maybe we can put the following line in a finally clause:
```
if (isRunning) {
        timeoutId = setTimeout(sendTransaction, pollingInterval);
      }
```

Reply:
yes you're right thanks Jones :)